Mathlib/Topology/Separation.lean

```diff
@@ -912,6 +914,287 @@ theorem TopologicalSpace.subset_trans {s t : Set X} (ts : t ⊆ s) :
   (embedding_inclusion ts).induced
 #align topological_space.subset_trans TopologicalSpace.subset_trans
 
+/-! ### R₁ (preregular) spaces -/
+
+section R1Space
+
+/-- A topological space is called a *preregular* (a.k.a. R₁) space,
+if any two topologically distinguishable points have disjoint neighbourhoods. -/
+@[mk_iff r1Space_iff_specializes_or_disjoint_nhds]
+class R1Space (X : Type*) [TopologicalSpace X] : Prop where
+  specializes_or_disjoint_nhds (x y : X) : Specializes x y ∨ Disjoint (𝓝 x) (𝓝 y)
+
+export R1Space (specializes_or_disjoint_nhds)
+
+variable [R1Space X] {x y : X}
+
+theorem disjoint_nhds_nhds_iff_not_specializes : Disjoint (𝓝 x) (𝓝 y) ↔ ¬x ⤳ y :=
+  ⟨fun hd hspec ↦ hspec.not_disjoint hd, (specializes_or_disjoint_nhds _ _).resolve_left⟩
+#align disjoint_nhds_nhds_iff_not_specializes disjoint_nhds_nhds_iff_not_specializes
+
+theorem specializes_iff_not_disjoint : x ⤳ y ↔ ¬Disjoint (𝓝 x) (𝓝 y) :=
+  disjoint_nhds_nhds_iff_not_specializes.not_left.symm
+
+/-- In an R₁ space, the `Specializes` relation is symmetric, i.e., an R₁ is an R₀ space. -/
+theorem Specializes.symm (h : x ⤳ y) : y ⤳ x := by
+  simpa only [specializes_iff_not_disjoint, disjoint_comm] using h
+#align specializes.symm Specializes.symm
+
+/-- In an R₁ space, the `Specializes` relation is symmetric, i.e., an R₁ is an R₀ space. -/
+theorem specializes_comm : x ⤳ y ↔ y ⤳ x := ⟨Specializes.symm, Specializes.symm⟩
+#align specializes_comm specializes_comm
+
+/-- In an R₁ space, the `Specializes` is equivalent to `Inseparable`, i.e., an R₁ is an R₀ space. -/
```

Comment:
```suggestion
/-- In an R₁ space, `Specializes` is equivalent to `Inseparable`. -/
```

Reply:
Why do you suggest removing the "i.e." parts? I added them on purpose: if we ever decide to add R₀ spaces (i.e., if we start caring about spaces which are R₀ but not R₁ for some reason), then one can immediately see the lemmas that need to be generalized from R₁ to R₀.

Reply:
Fine then.